src/app/Library/CrudPanel/Traits/Buttons.php

```diff
@@ -77,6 +77,11 @@ public function addButton($stack, $name, $type, $content, $position = false, $re
         return new CrudButton($name, $stack, $type, $content, $position);
     }
 
+    public function addCrudButton(CrudButton $crudButton)
+    {
+        $this->setOperationSetting('buttons', $this->buttons()->push($crudButton));
+    }
```

Comment:
Oh so we have a method to add a CrudButton directly, using the object? Nice! Good name too.